File: Data Structures & Algorithms/jump-game/submission-2.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        dp = [False]*n
        dp[n-1] = True

        for i in range(n-2, -1, -1):
            for j in range(1, nums[i]+1):
                dp[i] = dp[j+i] if j+i < n else True
                if dp[i]:
                    break
        
        return dp[0]

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut i = 0;
        let mut last = -101;

        for r in 0..nums.len() {
            if nums[r] != last {
                nums[i] = nums[r];
                i += 1;
                last = nums[r];
            }
        }

        i as i32
    }
}
